core/session.ts

```diff
@@ -190,17 +193,13 @@ export class Session {
     if (action instanceof test.Test) {
       return action;
     }
-
-    if (!(action instanceof Operation) || action.proto.hasOutput) {
-      const finalSchema =
-        actionOptions.sqlxConfig.schema ||
-        (actionOptions.sqlxConfig.type === "assertion"
-          ? this.config.assertionSchema
-          : this.config.defaultSchema);
-      action.proto.target = this.target(actionOptions.sqlxConfig.name, finalSchema);
-    } else {
-      delete action.proto.target;
-    }
+    const finalSchema =
```

Comment:
i have a suspicion that these lines that fix `target` and `name` for overridden `schema` settings are no longer needed after your https://github.com/dataform-co/dataform/pull/368. can you remove these lines and check the tests still pass?

Reply:
Perhaps I should have mentioned this on Slack before implementing it but this line is here because I added the ability to refer to datasets in tests using `Resolvable`. Without this, datasets in tests cannot refer to datasets in arbitrary schemas. See test case `bigquery_language_v2/definitions/example_test_case_fq_ref.sqlx` in this same CR:

````
config {
    type: "test",
    dataset: {schema: "df_integration_test", name: "example_table"}
}
````

In fact I believe we will need to add the ability to use `Resolvable`'s in other places like `dependencies`, `preOps`, `postOps`, etc. - Maybe I should create an issue or something?

Reply:
Can you clarify why this is required for tests to reference `Resolvable` datasets?

As far as I understand it:
- a dataset action registers itself with the `session`, possibly overriding its `schema` and/or `name`
- a test references a dataset, using a `Resolvable`
- at test `compile()`-time, the test can look up all actions in the `session`, searching for a dataset which matches the `Resolvable`

I guess I'm not clear on what these lines do that are required to make that work!

Reply:
Sorry Ben, that was a mistake from my part. This has nothing to do with the `Resolvable` on `Test`, however as you correctly point out if we take out this line we get an error on this test case:

`api.spec.ts`, `L849`:
`expect(exampleUsingOverriddenSchema).to.not.be.undefined;`

I also think that this smells bad and the override schema should already been captured.